File: Golang/BinaryTreeLevelOrderTraversal/BinaryTreeLevelOrderTraversal_test.go
package BinaryTreeLevelOrderTraversal

import (
	"testing"
	"reflect"
)

func TestLevelOrder(t *testing.T) {

	if len(levelOrder(nil)) == 0 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(levelOrder(
		&TreeNode{
			Val: 1,
		}), [][]int{{1,},}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(levelOrder(
		&TreeNode{
			Val: 3,
			Left: &TreeNode{
				Val: 9,
			},
			Right: &TreeNode{
				Val: 20,
				Left: &TreeNode{
					Val: 15,
				},
				Right: &TreeNode{
					Val: 7,
				},
			},
		}), [][]int{{3,}, {9, 20,}, {15, 7,},}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/BinaryTreeLevelOrderTraversalTest.py
from unittest import TestCase

from BinaryTreeLevelOrderTraversal import BinaryTreeLevelOrderTraversal, TreeNode


class TestBinaryTreeLevelOrderTraversal(TestCase):
    def test_levelOrder(self):
        b = BinaryTreeLevelOrderTraversal()

        self.assertTrue(len(b.levelOrder(None)) == 0)

        node0 = TreeNode(1)
        self.assertEqual(b.levelOrder(node0), [[1]])

        node1 = TreeNode(3)
        node1.left = TreeNode(9)
        node1.right = TreeNode(20)
        node1.right.left = TreeNode(15)
        node1.right.right = TreeNode(7)

        self.assertEqual(b.levelOrder(node1), [[3], [9, 20], [15, 7]])
